python/rmm/_lib/memory_resource.pyx

```diff
@@ -167,101 +163,70 @@ cdef class FixedSizeMemoryResource(MemoryResource):
+        specified allocation size.
+
+        Call add_bin to add fixed-size bin allocators.
 
         Parameters
         ----------
-        upstream : MemoryResource
-            The upstream memory resource used to allocate pools of blocks.
-        size_base : int, optional
-            The base of allocation block sizes (defaults is 2).
-        min_size_exponent : int, optional
-            The exponent of the minimum fixed block size to allocate
-            (default is 18).
-        max_size_exponent : int, optional
-            The exponent of the maximum fixed block size to allocate
-            (default is 22).
-        initial_blocks_per_size : int, optional
-            The number of blocks to preallocate from the upstream memory
-            resource, and to allocate when all current blocks are in use.
+        upstream_mr : MemoryResource
+            The memory resource to use for allocations larger than any of the
+            bins
         """
         pass
 
-
-cdef class HybridMemoryResource(MemoryResource):
-    def __cinit__(
+    cpdef add_bin(
         self,
-        MemoryResource small_alloc_mr,
-        MemoryResource large_alloc_mr,
-        size_t threshold_size=1<<22
+        size_t allocation_size,
+        object bin_resource=None
     ):
-        self.c_obj.reset(
-            new thread_safe_resource_adaptor_wrapper(
-                shared_ptr[device_memory_resource_wrapper](
-                    new hybrid_memory_resource_wrapper(
-                        small_alloc_mr.c_obj,
-                        large_alloc_mr.c_obj,
-                        threshold_size
-                    )
-                )
-            )
-        )
+        """
+        Adds a bin of the specified maximum allocation size to this memory
+        resource. If specified, uses bin_resource for allocation for this bin.
+        If not specified, creates and uses a FixedSizeMemoryResource for
+        allocation for this bin.
 
-    def __init__(
-        self,
-        MemoryResource small_alloc_mr,
-        MemoryResource large_alloc_mr,
-        size_t threshold_size=1<<22
-    ):
-        """"
-        Allocates memory from one of two allocators based on the requested
-        size.
+        Allocations smaller than allocation_size and larger than the next
+        smaller bin size will use this fixed-size memory resource.
 
         Parameters
         ----------
-        small_alloc_mr : MemoryResource
-            The memory resource to use for small allocations.
-        large_alloc_mr : MemoryResource
-            The memory resource to use for large allocations.
-        threshold_size : int, optional
-            Size in bytes representing the threshold beyond which
-            `large_alloc_mr` is used.
+        allocation_size : size_t
+            The maximum allocation size in bytes for the created bin
+        bin_resource : shared pointer to device_memory_resource_wrapper
```

Comment:
Isn't `bin_resource` an instance of `MemoryResource` _from_ which we grab the shared pointer? Python users shouldn't have to know anything about shared pointers

Reply:
Probably. I still find Cython crazy and convoluted.

Reply:
So the fix I'm suggesting is basically just the following:

```suggestion
        bin_resource : MemoryResource
```

Reply:
Basically just done. Thanks!